workflows/gtfs_validator_execution.yml

```diff
@@ -0,0 +1,187 @@
+main:
+  params: [event]
+  steps:
+      - log_event:
+          call: sys.log
+          args:
+              text: ${event}
+              severity: INFO
+      - extract_data:
+          assign:
+          - data: ${event.data.protoPayload}
+      - init:
+          assign:
+            - datasetsBucket: ${sys.get_env("datasets_bucket_name")}
+            - reportsBucket: ${sys.get_env("reports_bucket_name")}
+            - validatorEndpoint: ${sys.get_env("validator_endpoint")}
+            - feedID: ${text.split(data.resourceName, "/")[5]}
+            - datasetID: ${text.split(data.resourceName, "/")[6]}
+            - url: ${"https://storage.googleapis.com/" + datasetsBucket + "/" + feedID + "/" + datasetID + "/" + datasetID + ".zip"}
+            - endpoint: ${validatorEndpoint + "/create-job"}
```

Comment:
I can't see the run-validator call. Don't we need to call this to do the actual validation?

Reply:
Not that I'm aware of. From reviewing the UI, we have two scenarios for handling uploads:

- **Scenario 1: User uploads a zip file.** We call an endpoint to create a job, then use another endpoint to upload the zip via the URL provided in the job creation response. The validator runs once the file is uploaded.

- **Scenario 2: User provides a URL.** This URL is directly used in the POST request to create the job, streamlining the process.

We're using Scenario 2 in this case, which directly utilizes the provided URL for job creation.

Reply:
The validation is triggered when a file is placed in the right GCP bucket. If the URL is provided, the create-job places the downloaded file in the bucket. When the user uploads the file, the create-job returns a signed URL, and the client is responsible for uploading the file to the bucket.

Reply:
I see. It's all done with a notification from the user-uploads bucket. The pubsub subscription calls the run-validator endpoint automatically.